TestDeletePods {cmd}

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-test-go/18465/

Failed: TestDeletePods {cmd}

```
panic: runtime error: invalid memory address or nil pointer dereference
/usr/local/go/src/runtime/panic.go:500 +0x1ae
/usr/local/go/src/testing/testing.go:579 +0x474
/usr/local/go/src/runtime/panic.go:458 +0x271
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/util/printing.go:60 +0x6e
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/drain.go:426 +0x2ef
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:213 +0x7e
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:168 +0xa7
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:185 +0x93
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:174 +0x5f
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/drain.go:440 +0x171
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/drain_test.go:628 +0x33e
/usr/local/go/src/testing/testing.go:610 +0xca
/usr/local/go/src/testing/testing.go:646 +0x530
```

--- a/pkg/kubectl/cmd/drain.go
+++ b/pkg/kubectl/cmd/drain.go
@@ -54,7 +54,6 @@ type DrainOptions struct {
 	nodeInfo           *resource.Info
 	out                io.Writer
 	typer              runtime.ObjectTyper
-	ifPrint            bool
 }
 
 // Takes a pod and returns a bool indicating whether or not to operate on the
@@ -197,8 +196,6 @@ func (o *DrainOptions) SetupDrain(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-
-	o.ifPrint = true
 
 	r := o.factory.NewBuilder().
 		NamespaceParam(cmdNamespace).DefaultNamespace().
@@ -422,9 +419,7 @@ func (o *DrainOptions) waitForDelete(pods []api.Pod, interval, timeout time.Dura
 		for i, pod := range pods {
 			p, err := getPodFn(pod.Namespace, pod.Name)
 			if apierrors.IsNotFound(err) || (p != nil && p.ObjectMeta.UID != pod.ObjectMeta.UID) {
-				if o.ifPrint {
-					cmdutil.PrintSuccess(o.mapper, false, o.out, "pod", pod.Name, false, "deleted")
-				}
+				cmdutil.PrintSuccess(o.mapper, false, o.out, "pod", pod.Name, false, "deleted")
 				continue
 			} else if err != nil {
 				return false, err

--- a/pkg/kubectl/cmd/drain_test.go
+++ b/pkg/kubectl/cmd/drain_test.go
@@ -621,9 +621,11 @@ func TestDeletePods(t *testing.T) {
 		},
 	}
 
-	o := DrainOptions{}
-	o.ifPrint = false
 	for _, test := range tests {
+		f, _, _, _ := cmdtesting.NewAPIFactory()
+		o := DrainOptions{}
+		o.mapper, _ = f.Object()
+		o.out = os.Stdout
 		_, pods := createPods(false)
 		pendingPods, err := o.waitForDelete(pods, test.interval, test.timeout, test.getPodFn)
 

